(DOCSP-14219): Add new API endpoint to openAPI docs, add description to Authenticate Client Requests page (#813)
mongodb/docs-realm: source/reference/authenticate-http-client-requests.txt

      curl --location --request POST 'https://realm.mongodb.com/api/client/v2.0/auth/session' \
        --header 'Content-Type: application/json' \
        --header 'Authorization: Bearer <Refresh Token>'

.. _verify-client-access-token:

Verify a Client Access Token
----------------------------

If you want to verify that a client access token is valid, you can do so using
the verify access token endpoint in the :ref:`administration API <admin-api>`.

The Admin API verify access token endpoint accepts a ``POST`` request that
includes these details:

> - Use the client access token you got using one of the above methods for the payload

Reviewer: ```suggestion
- the client access token you got using one of the :ref:`above methods <get-client-api-access-token>` for the payload
```

> - You'll need an :ref:`Admin API access token <realm-api-authentication>` ``Authorization`` header

Reviewer: ```suggestion
- an :ref:`Admin API access token <realm-api-authentication>` ``Authorization`` header
```

> - The {groupId} and {appId} form the URL. View :ref:`the admin API documentation <realm-api-project-and-application-ids>` for details on how to find these

Reviewer: ```suggestion
- the :ref:`{groupId} and {appId} <realm-api-project-and-application-ids>` for your backend {+app+}
```

With this information, call this API endpoint to verify that the client access 
token is valid:

.. code-block:: text
   
   https://realm.mongodb.com/api/admin/v3.0/​groups/​{groupId}/​apps/​{appId}/​users/​verify_token

.. important::

>    Because this endpoint requires admin API access, implementing this is best 

Reviewer: We tend to add an extra newline after note directives like this, so I would suggest adding that for consistency (though it is not required, as your staging output shows). I would also suggest adding a title, maybe something like "Do Not Access the Admin API Directly from Client Applications"

Author: Amusingly, I had tried adding a title before, but it showed up in the body of the directive as regular text, so I'm guessing that was related to not having the newline. I'll add the newline and put in a title again 👍

>    done as a backend service behind an authorization layer - not directly 

Reviewer: For some reason DevEd seldom uses dashes in our documentation. But I'm inclined to say that this point is important enough to draw attention to in this way.

   in the client app.

.. example::

>    The following request demonstrates how to use a the Admin API endpoint to 

Reviewer: ```suggestion
   The following request demonstrates how to use the Admin API endpoint to 
```

   verify that a user's client access token is valid. Replace ``<API Access Token>`` 
   with the ``access_token`` value you get after authenticating with the
   Admin API, and ``<Client Access Token>`` with the client access token you

>    got using one of the above methods. 

Reviewer: I would also mention that users need to replace the groupId and appId placeholders here with the values obtained from their Realm app, probably with another link to the admin api docs on those concepts.

   
   .. code-block:: shell
      :emphasize-lines: 3
   
      curl --request POST \
        --url https://realm.mongodb.com/api/admin/v3.0/groups/{groupId}/apps/{appId}/users/verify_token \
        --header 'Authorization: Bearer <API Access Token>' \
        --header 'Content-Type: application/json' \

>         --data '{ "token": "<Client Access Token>" }'

Reviewer: Finally, I would add a sentence after this explaining that if you get a 200 OK, the token is valid, otherwise it is invalid. And maybe mention the type of response you get if your admin auth token is expired/invalid, since that is a pretty common thing to hit.

Author: I'm glad you mentioned this! Turns out, you get a 200 OK if the token is valid, OR if the token is expired, but the response body is different in those cases. I'll cover that here, and maybe add a note in the openAPI doc about it, too.

      curl --location --request POST 'https://realm.mongodb.com/api/client/v2.0/auth/session' \
        --header 'Content-Type: application/json' \
        --header 'Authorization: Bearer <Refresh Token>'

.. _verify-client-access-token:

Verify a Client Access Token
----------------------------

If you want to verify that a client access token is valid, you can do so using
the `verify access token endpoint <{+base-url+}{+admin-api-page+}post-/groups/{groupId}/apps/{appId}/users/verify-token>`__ 
in the administration API.

> This endpoint accepts ``POST`` request that includes these details:

Reviewer 2: ```suggestion
This endpoint accepts a ``POST`` request that includes these details:
```

.. important:: Do Not Access the Admin API Directly from Client Applications
   
   Because this endpoint requires admin API access, implementing this is best 
   done as a backend service behind an authorization layer - not directly 
   in the client app.

.. example::

   The following request demonstrates how to use the Admin API endpoint to 
   verify that a user's client access token is valid. Replace ``<API Access Token>`` 
   with the ``access_token`` value you get after authenticating with the
   Admin API, and ``<Client Access Token>`` with the client access token you
   got using one of the above methods. In the URL, replace {groupId} and {appId}
   with :ref:`the values for your Realm app <realm-api-project-and-application-ids>`.
   
   .. code-block:: shell
      :emphasize-lines: 3
   
      curl --request POST \
        --url https://realm.mongodb.com/api/admin/v3.0/groups/{groupId}/apps/{appId}/users/verify_token \
        --header 'Authorization: Bearer <API Access Token>' \
        --header 'Content-Type: application/json' \
        --data '{ "token": "<Client Access Token>" }'

.. _verify-client-access-token-responses:

Responses
~~~~~~~~~

> **Valid access token**: If you get a ``200 OK`` with this object, the <Client Access Token> is valid:

Reviewer: ```suggestion
**Valid access token**: If you get a ``200 OK`` with a response that contains the following fields, the <Client Access Token> is valid:
```

When I first read this, I translated it to "if you get a 200 OK, it's valid" in my head, which the second bullet point contradicts. I figure if that can happen to me, it can happen to any reader, so we should be a tiny bit more specific here than "this object".
